core/genesis.go

```diff
@@ -240,12 +240,20 @@ func SetupGenesisBlock(db ethdb.Database, genesis *Genesis) (*params.ChainConfig
 
 	// Check config compatibility and write the config. Compatibility errors
 	// are returned to the caller unless we're already at block zero.
-	headBlock := rawdb.ReadHeadBlock(db)
-	if headBlock == nil {
-		return newcfg, fmt.Errorf("missing head block")
+	// headBlock := rawdb.ReadHeadBlock(db)
+	// if headBlock == nil {
+	// 	return newcfg, fmt.Errorf("missing head block")
+	// }
+
+	// we use last accepted block for cfg compatibility check to handle
+	// retroactive upgrades gracefully.
```

Comment:
```suggestion
	// we use last accepted block for cfg compatibility check. Note this allows
	// the node to continue making progress if it had halted previously due to
	// a missing upgrade configuration.
```
or something like that?